demo-app/src/components/aruba/ArubaMap.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Box } from 'grommet';
+import { Grommet } from 'grommet-icons';
+import {
+  Cluster,
+  Controls,
+  Map,
+  Marker,
+  MarkerCluster,
+  Pin,
+} from 'grommet-leaflet-core';
+
+import { userLocation, formatLocationsToLatLng } from '../../utils/locations';
+import { generateArubaLocations } from './ArubaData';
+import { ClusterPopup } from '../../ClusterPopup';
+import { ArubaPin } from './ArubaPin';
+import { ArubaPopup } from './ArubaPopup';
+import { hpeLeaflet } from '../../themes';
+import { getClusterStatus, getClusterSize } from '../../utils/status';
+
+function ArubaMap() {
+  const [geolocation, setGeolocation] = useState();
+  const [locations, setLocations] = useState();
+  const containerRef = useRef();
+  const mapContainerRef = useRef();
+
+  // get the user's location
+  useEffect(() => {
+    userLocation().then(location => {
+      setGeolocation(location);
+      setLocations(generateArubaLocations(20, { center: location, radius: 6 }));
+    });
+  }, []);
+
+  return (
+    <Box ref={containerRef} flex background="background-contrast">
+      {geolocation && (
+        <Map
+          id="map"
+          ref={mapContainerRef}
+          center={geolocation}
+          zoom={6}
+          zoomControl={false}
```

Comment:
Let's try to minimize the number props specified here. I believe these three can be removed b/c Map now has a greater number of defaults.

Reply:
Done